### PR TITLE
Issue/11237 extract top performers view

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Part of: #11237 

### Description

This PR adds the following changes: 
- Converts top performers' view to full composable code
- Adopts new implementation to handle all events 
- Extracts any top performers-related logic out of DashboardViewModel

This PR doesn't adjust the composable code to the recent designs published by Joe. Those came out after I had already worked on this. I will address that in a different PR. 

### Testing instructions

1. Enable `DYNAMIC_DASHBOARD` feature flag
2. Launch the app
3. From my store tab check top performers are loaded correctly and skeletons are shown
4. Click on a top performer item and check it opens the product detail
5. Load a custom range where there's not sales and check the empty case is displayed

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/2663464/0fe6b796-1580-407d-bef0-1d2d847ef666

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          4 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>DashboardTopPerformersViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>TopPerformersState</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/ProductThumbnail.kt [hunk 1]
```diff
@@ -34,7 +33,7 @@ fun ProductThumbnail(
         contentDescription = contentDescription,
         contentScale = ContentScale.Crop,
         modifier = modifier
-            .size(dimensionResource(id = dimen.major_275))
-            .clip(shape = RoundedCornerShape(dimensionResource(id = dimen.minor_100)))
+            .size(42.dp)
+            .clip(shape = RoundedCornerShape(6.dp))
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/ProductThumbnail.kt]: Why this change?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/topperformers/DashboardTopPerformersCard.kt [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+}
+
+@Composable
+private fun TopPerformersCard(
+    topPerformersState: TopPerformersState?,
+    lastUpdateState: String?
+) {
+    Card(
+        modifier = Modifier
+            .padding(top = 16.dp, bottom = 16.dp)
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/topperformers/DashboardTopPerformersCard.kt:96]: Related to my above comment, was there a team decision to use literal dimensions here instead of using the ones from the resources? Sorry if I missed it.

**Review by hichamboushaba — COMMENTED** [review]
Nice work @JorgeMucientes.

I left a question about the usage of literal dimensions in Compose code, please check it out.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/topperformers/DashboardTopPerformersCard.kt (hunk 2 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/topperformers/DashboardTopPerformersCard.kt:96]: Hey @hichamboushaba good question. Technically there hasn't been an "official" debate about this with all the team involved. But a few months back there was a discussion about it in #mobile-woo-async-mtg channel: p1700739951209439-slack-CGPNUU63E
Up until that point I had been using `dimensionresource()` everywhere. But after reading that and seeing that other teams are using dimens directly, I agreed with the points stated in the discussion: 
- The current defined dimens in XML are not semantic and the same dimen is used for many different things in the UI. In case of a migration, all usages will have to be revisited manually, loosing the clearest benefit of reusing defined dimens. 
- Since, there's not a clear benefit of using defined dimens anymore, I believe using `dimensionresource()` is extra boilerplate and makes it less clear when reading compose code which value is being used (since dimens names are not semantic) 

Let me know your thoughts :). If you disagree lets reopen the debate involving the whole team.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/ProductThumbnail.kt (hunk 1 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/ProductThumbnail.kt]: Setting the value to `6.dp` instead of prior value of `8.dp` to match the design better. Additionally, I decided it was clearer to use the values directly instead of existing `dimens` values for the reasons stated below.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/topperformers/DashboardTopPerformersCard.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/topperformers/DashboardTopPerformersCard.kt:96]: Thanks @JorgeMucientes for the clarification.
Personally, I value consistency for those points more than anything else, I don't see a big difference between the two patterns, I would suggest you restart the debate if you feel the points shared on the other discussion are this important, and then have the team decide on a single approach, otherwise we'll end up with Compose functions where both patterns are used, and it won't be pretty IMHO.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/ProductThumbnail.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/ProductThumbnail.kt]: In figma I see that the radius is `4.dp`, so we should probably still reduce it further.

Screenshot 2024-04-21 at 15 01 38

**Review by hichamboushaba — APPROVED** [review]
I thought I already approved this in my previous review, I'm approving it now, feel free to merge regardless of your decision regarding the pending discussions, nice work 💯.
